app/controllers/lists_controller.rb

```diff
@@ -4,9 +4,15 @@ class ListsController < ApplicationController
   def create
     @list = current_user.lists.new(name: params[:name])
     if @list.save
-      redirect_to root_path
+      respond_to do |format|
+        format.html { redirect_to root_path(), notice: "List created" }
+        format.json { render json: { message: "List created", id: @list.id } }
+      end
     else
-      redirect_to root_path, notice: "Invalid list name"
+      respond_to do |format|
+        format.html { redirect_to root_path(), notice: "Invalid list name" }
```

Comment:
Same but as above regarding parens.
```suggestion
        format.html { redirect_to root_path, notice: "Invalid list name" }
```